Adjust formula to work on High Sierra, and not have a name clash

justinclift opened the pull request:
Removing the gcc check seems to be needed, otherwise this error occurs:

```bash
$ brew tap tinygo-org/homebrew-tools
Updating Homebrew...
==> Tapping tinygo-org/tools
Cloning into '/usr/local/Homebrew/Library/Taps/tinygo-org/homebrew-tools'...
remote: Enumerating objects: 4, done.
remote: Counting objects: 100% (4/4), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 4 (delta 0), reused 2 (delta 0), pack-reused 0
Unpacking objects: 100% (4/4), done.
Error: Invalid formula: /usr/local/Homebrew/Library/Taps/tinygo-org/homebrew-tools/llvm.rb
Calling fails_with :gcc_4_2 is disabled! There is no replacement.
Please report this to the tinygo-org/tools tap:
  /usr/local/Homebrew/Library/Taps/tinygo-org/homebrew-tools/llvm.rb:121

Error: Cannot tap tinygo-org/tools: invalid syntax in tap!
$
```

Renaming the formula to llvm7 seems sensible too, for avoiding a name clash with the Homebrew-core llvm formula this was based upon.

deadprogram commented:
Thanks for this, @justinclift 

Is anyone else able to confirm it works as expected?

justinclift commented:
Come to think of it, we have macOS Sierra installed on a separate drive too (same physical box though).  I'll try this out on that as well, just to make sure no weirdness shows up.

justinclift commented:
It's running now.  It took ~188mins to compile on High Sierra, so it'll probably take roughly similar on this.

justinclift commented:
The build on macOS Sierra also succeeded. :smile:

With Homebrew, it prints a useful info notice ("Caveats") after the installation of packages.  This is what's shown for the `brew tap` command (note the "keg-only" piece for libffi):

```
==> libffi
libffi is keg-only, which means it was not symlinked into /usr/local,
because some formulae require a newer version of libffi.

For compilers to find libffi you may need to set:
  export LDFLAGS="-L/usr/local/opt/libffi/lib"

==> llvm7
To use the bundled libc++ please add the following LDFLAGS:
  LDFLAGS="-L/usr/local/opt/llvm7/lib -Wl,-rpath,/usr/local/opt/llvm7/lib"

llvm7 is keg-only, which means it was not symlinked into /usr/local,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

If you need to have llvm7 first in your PATH run:
  echo 'export PATH="/usr/local/opt/llvm7/bin:$PATH"' >> ~/.bash_profile

For compilers to find llvm7 you may need to set:
  export LDFLAGS="-L/usr/local/opt/llvm7/lib"
  export CPPFLAGS="-I/usr/local/opt/llvm7/include
```

This may explain the missing `-lffi` - as mentioned in the Slack channel - when running llvm-config.  Will investigate later.

deadprogram commented:
From those messages it would appears that the developer will also need to make those changes to the `PATH`. Perhaps we need to also provide a shell script so you can `source env.sh` or something like that to get the environment setup correctly when using the TinyGo compiler.

justinclift commented:
Possibly.  With "keg only" packages, there's a Homebrew command for creating the soft links in standard Homebrew places so they're picked up in the normal path structure:

    $ brew link <package name>

eg:

    $ brew list llvm7 --force

The `--force` option is sometimes useful, as there are some types of "keg only" packages Homebrew would rather not link in anyway.  But (from memory), even `--force` no longer works with some packages.  Will experiment later and see where things stand.

It may be better to remove the line marking the package as `keg only`, so it's instead installed normally, like other packages:

https://github.com/tinygo-org/homebrew-tools/blob/31365bce1439c5bc3e967364f48ada194cc897a7/llvm.rb#L101

The libffi formula may be a problem though, as we're not using our own version of it.  Yet. :wink:

deadprogram commented:
Since we are now installing under a different name, removing the `keg_only` should be safe, right?

justinclift commented:
Hmmm, I'd be inclined to remove the `keg_only` option anyway, just to keep things simple.

If people (installing this) didn't want it to be in their path (etc)... then why are they installing it? :smile:

deadprogram commented:
Agreed, then. Can you please remove and then update this PR with that commit?

justinclift committed:
Adjust formula to work on High Sierra, and not have a name clash

Also removes the 'keg_only' option, as it's unlikely to be useful
for our users.

justinclift commented:
Done. :smile:

deadprogram commented:
Thank you very much @justinclift this will be quite helpful. Now merging.
